sdk/src/main/java/org/torusresearch/customauth/CustomAuth.java

```diff
@@ -147,13 +157,16 @@ public CompletableFuture<TorusAggregateLoginResponse> triggerAggregateLogin(Aggr
             String aggregateTokenString = TextUtils.join(Character.toString((char) 29), aggregateIdTokenSeeds);
             String aggregateIdToken = Hash.sha3String(aggregateTokenString).substring(2);
             aggregateVerifierParams.setVerifier_id(aggregateVerifierId);
-            HashMap<String, Object> aggregateVerifierParamsHashMap = new HashMap<>();
-            aggregateVerifierParamsHashMap.put("verify_params", aggregateVerifierParams.getVerify_params());
-            aggregateVerifierParamsHashMap.put("sub_verifier_ids", aggregateVerifierParams.getSub_verifier_ids());
-            aggregateVerifierParamsHashMap.put("verifier_id", aggregateVerifierParams.getVerifier_id());
-            return this.getTorusKey(aggregateLoginParams.getVerifierIdentifier(), aggregateVerifierId, aggregateVerifierParamsHashMap, aggregateIdToken).thenApply((torusKey) -> Pair.create(userInfoArray, torusKey));
+            VerifierParams aggregateVerifierParamsHashMap = new VerifierParams(aggregateVerifierParams.getVerifier_id(), null,
+                    aggregateVerifierParams.getSub_verifier_ids(), aggregateVerifierParams.getVerify_params());
+            try {
+                TorusKey torusKey = this.getTorusKey(aggregateLoginParams.getVerifierIdentifier(), aggregateVerifierId, aggregateVerifierParamsHashMap, aggregateIdToken);
+                return CompletableFuture.completedFuture(Pair.create(userInfoArray, torusKey));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
         }).thenApplyAsync(pair -> {
-            TorusKey torusKey = pair.second;
+            TorusKey retrieveSharesResponse = pair.second;
```

Comment:
Think the name here was correct, we're not getting a share back but the final key
```suggestion
            TorusKey torusKey = pair.second; 
```